plugins/notifications-backend-module-slack/src/lib/util.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  NotificationPayload,
+  NotificationSeverity,
+} from '@backstage/plugin-notifications-common';
+import { ChatPostMessageArguments, KnownBlock } from '@slack/web-api';
+
+export function toChatPostMessageArgs(options: {
+  channel: string;
+  payload: NotificationPayload;
+}): ChatPostMessageArguments {
+  const { channel, payload } = options;
+
+  const args: ChatPostMessageArguments = {
+    channel,
+    text: payload.title,
+    attachments: [
+      {
+        color: getColor(payload.severity),
+        blocks: toSlackBlockKit(payload),
+        fallback: payload.title,
+      },
+    ],
+  };
+
+  return args;
+}
+
+export function toSlackBlockKit(payload: NotificationPayload): KnownBlock[] {
+  const { description, link, severity, topic } = payload;
+  return [
+    {
+      type: 'section',
+      text: {
+        type: 'mrkdwn',
+        text: description ?? 'No description provided',
+      },
+      accessory: {
```

Comment:
If there's no link, should this be discarded? Don't know the slack api but just thinking empty string as url is interesting